backstory/src/test/java/com/google/sps/servlets/AnalyzeImageServletTest.java

```diff
@@ -0,0 +1,178 @@
+import com.google.sps.story.StoryManagerImpl;
+import com.google.sps.servlets.AnalyzeImageServlet;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.sps.servlets.data.BlobstoreManagerFactory;
+import com.google.sps.servlets.data.BackstoryDatastoreServiceFactory;
+import com.google.sps.servlets.data.EntityFactory;
+import com.google.sps.servlets.data.ImagesManagerFactory;
+import com.google.sps.servlets.data.StoryManagerFactory;
+import com.google.sps.servlets.data.BackstoryUserServiceFactory;
+import com.google.sps.servlets.data.StoryAnalysisManagerFactory;
+import com.google.sps.servlets.data.BlobstoreManager;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.User;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.ArgumentCaptor;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Map;
+import java.util.HashMap;
+import com.google.sps.perspective.data.APINotAvailableException;
+import com.google.sps.perspective.data.NoAppropriateStoryException;
+import com.google.sps.perspective.data.StoryDecision;
+
+/**
+ * Tests for the analyze image servlet, which contains all image analysis and backstory generation
+ * functionalities.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public final class AnalyzeImageServletTest {
+
+  /**
+   * Tests that the correct input is passed from the initial call to the managers, 
+   * and that the correct ouput is passed from the managers to the permanent storage service.
+   */
+  @Test
+  public void testDoPostInputAndOuput() throws APINotAvailableException, NoAppropriateStoryException, IOException {
+    HttpServletRequest mockRequest = mock(HttpServletRequest.class);
+    HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+
+    // 
+    AnalyzeImageServlet servlet = new AnalyzeImageServlet();
+    ImagesManager mockImagesManager = mock(VisionImagesManager.class);
+    StoryManager mockStoryManager = mock(StoryManagerImpl.class);
+    StoryAnalysisManager mockStoryAnalysisManager = mock(PerspectiveStoryAnalysisManager.class);
+    DatastoreService mockDatastoreService = mock(DatastoreService.class);
+    Entity mockAnalyzedImageEntity = mock(Entity.class);
+    BlobstoreManager mockBlobstoreManager = mock(BlobstoreManager.class);
+    UserService mockUserService = mock(UserService.class);
+
+    String userEmail = "user@gmail.com";
+    String userAuthenticationDomain = "authentication";
+    User testUser = new User(userEmail, userAuthenticationDomain);
+
+
+    // Creates the required real test objects, and sets the required behavior of the mocks.
+    boolean userIsLoggedIn = true;
+    when(mockUserService.isUserLoggedIn()).thenReturn(userIsLoggedIn);
```

Comment:
Since `userIsLoggedIn` is only used once, consider omitting the local variable and just inserting literal `true` here.

Reply:
Sure, updated.